Add Cardano.Wallet.Jormungandr.Network.Api

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->

#219 

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] Servant type in the .Api module, with Plain and Binary content types.
- [x] I added a FromBinary class in the binary module to play nicely with the Binary content-type


# Comments

<!-- Additional comments or screenshots to attach if any -->

Whoops :man_shrugging: 

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

parsonsmatt commented on test/integration/Test/Integration/Scenario/EosWallets.hs:
```diff
@@ -6,16 +6,19 @@ module Test.Integration.Scenario.EosWallets
         fixture  <- setup defaultSetup
         eowallet <- successfulRequest $ Client.postEosWallet $- NewEosWallet
-            (take 3 $ fixture ^. externallyOwnedAccounts)
+            (NE.fromList $ take 3 $ fixture ^. externallyOwnedAccounts)
```
`fromList` is partial, but this is a test fixture so it's OK

Anviking committed:
Add Cardano.Wallet.Jormungandr.Network.Api

- Servant type in the .Api module, with Plain and Binary content types.
- I added a FromBinary class in the binary module to play nicely with the Binary content-type
